File: src/main/java/net/matees/arcade/mobrush/settings/EnableHostileMobs.java
package net.matees.arcade.mobrush.settings;

import org.bukkit.Material;

import net.matees.settings.BooleanSetting;

public class EnableHostileMobs extends BooleanSetting {

    private boolean setting = true;

    private static final EnableHostileMobs INSTANCE = new EnableHostileMobs();

    private EnableHostileMobs() {
    }

    public static EnableHostileMobs getInstance() {
        return INSTANCE;
    }

    @Override
    public String getName() {
        return "Enable Hostile Mobs";
    }

    @Override
    public String getDescription() {
        return "Allows spawning of hostile mobs";
    }

    @Override
    public Boolean getSetting() {
        return setting;
    }

    @Override
    public void setSetting(Boolean setting) {
        this.setting = setting;
    }

    @Override
    public Material getMenuItemMaterial() {
        return Material.CREEPER_SPAWN_EGG;
    }

    @Override
    public int getMenuItemSlot() {
        return 13;
    }

}

File: src/main/java/net/matees/arcade/mobrush/settings/EnableHostileMobs.kt
package net.matees.arcade.mobrush.settings

import net.matees.settings.AbstractBooleanSetting
import org.bukkit.Material

class EnableHostileMobs private constructor() : AbstractBooleanSetting() {
    override var setting: Boolean? = true

    override val name: String
        get() = "Enable Hostile Mobs"

    override val description: String
        get() = "Allows spawning of hostile mobs"

    override val menuItemMaterial: Material
        get() = Material.CREEPER_SPAWN_EGG

    override val menuItemSlot: Int
        get() = 13

    companion object {
        val instance: EnableHostileMobs = EnableHostileMobs()
    }
}
